hip-0000-registry-contract-for-naming-services.md

```diff
@@ -0,0 +1,160 @@
+In the beginning, ICANN (The Internet Corporation for Assigned Names and Numbers) responsible for web2 domains had started with thirteen TLDs. Today, we have hundreds of gTLD (generic Top Level Domains) like .biz, .capital etc. At the same time, no two SLD within a TLD match or so to say, duplication of domain names within a TLDs does not exist across the globe although buying of domains is administered by hundreds of domain registrars, resellers etc.
+Riding on the exceptional capabilities of Hedera & following the design philosophy of ICANN where no two SLDs are identical, proposed default structured format of TLDs & SLDs in the form of registry contract within Hedera would allow multi-parties to co-exist without leading to the confusion. 
+This approach of proposed registry contract containing default structure format of TLDs & SLDs would possibly make Hedera a natural fit for decentralized ICANN allowing inter-chain dialogues too.
+
+## Rationale
+
+Taking the motivation, a further step ahead, it is proposed to allow multiple parties who could be resellers, companies to exist within the ecosystem of Hedera and allow community/users to mint/book domains ending in .hbar or any other TLD without the fear of duplication or phishing attempts.
+Proposed Hedera centric registry contract that would also be extended to other inter-operable chains in the future would be able to accommodate multiple parties. Same registry contract would contain the real truth of mappings between Account IDs on the Mainnet & the corresponding human readable user-names. For example, john.hbar tied to 0.0.XXXXXXX
+
+## Specification
+
+Architecture of the proposed registry contract over Hedera would be, as follows:
+
+![RegistryContract](https://user-images.githubusercontent.com/97507177/181780608-e6bc217b-ee21-4b6d-a9a4-a28916a5ec87.png)
+
+ 
+(Image source: https://drive.google.com/file/d/1e3yxtaY8glEiptSXlrNxznxfzTfk1SKr/view) 
+
+## Detailed note on the flow
+
+**1. What is Registry Contract Suite?**<br />
+Registry Contract Suite is a group of smart Contracts developed over Solidity. These smart contracts are intended to register HTLDs (Hedera Top-Level-Domains) and Second-Level-Domains (SLDs) in a hierarchical manner. Domains and HTLDs once registered could be queried and looked up to avoid duplication.<br />
+**2. How does Registry Contract Suite Works?**<br />
+Registry Contract Suite is an intelligent group of smart contract that maintains an alphabetical hierarchy to register HTLDs and is smart enough to create and store HTLDs in their respective Alphabetical Registry Component.<br />
+	Once a HTLD is registered by the provider, an entry in respective alphabetical hierarchy smart contract is made. All further domains and subdomains under that hierarchy would be stored in that structure. So that for future, queries for that registered domain names or HTLD/s could be appropriately handled.<br />
```

Comment:
I'm not sure that an alphabetical hierarchy of smart contracts is necessary, can there be a subsection defining this?